### DISCUSSION TITLE
Asking about edumeet strcuture

### DISCUSSION THREAD

**Discussion by saif-o99** [edumeet/edumeet-room-server#88] [General]
Hey guys, 
im seeing that every event action is contrlled by the client. for example moderator actions like mute peer
the client sends an event to room server "moderator:mute" then the room server sends an event to the client telling it to mute the producer from the client side.
while we could stop the producer from the room server directly.
is there a reason on why the structure is implemented like this?

thanks

**Comment by havfo** [comment]
There is no "permanent state" tied to the case of being muted by the moderator, so the muted party is free to unmute themselves again. Because of that we have seen no need to enforce it more on the server side. This also keeps the logic simpler in that the client itself is the one always in control of the mute/unmute state of its producer.
